libgnucash/engine/qofevent.cpp

```diff
@@ -43,26 +56,17 @@ static QofLogModule log_module = QOF_MOD_ENGINE;
 static gint
 find_next_handler_id(void)
 {
-    HandlerInfo *hi;
-    gint handler_id;
-    GList *node;
-
     /* look for a free handler id */
-    handler_id = next_handler_id;
-    node = handlers;
+    auto handler_id = next_handler_id;
 
-    while (node)
+ restart:
+    for (const auto& hi : handlers)
     {
-        hi = static_cast<HandlerInfo*>(node->data);
-
-        if (hi->handler_id == handler_id)
+        if (hi.handler_id == handler_id)
```

Comment:
This will never happen: `handler_id` is inited to `next_handler_id` that's incremented every time this function is called and never decremented, so there can never be an entry in the vector with that value.